### PR TITLE
asm.calls final tweakings / bug fixes

### PR DESCRIPTION
**Pull request by oddcoder**

### CODE REVIEW

File: libr/core/disasm.c [hunk 1]
```diff
@@ -2837,13 +2864,17 @@ static void ds_print_calls_hints(RDisasmState *ds) {
...
-			r_anal_type_func_args_name (anal, name, i),
-			i == arg_max - 1 ? ");\n": ",");
-		free (type);
+	if (arg_max == 0) {
+		r_cons_printf("void);\n");
+	} else {
+		for (i = 0; i < arg_max; i++) {
+			char *type = r_anal_type_func_args_type (anal, name, i);
+			r_cons_printf ("%s%s%s%s%s", i == 0 ? "": " ", type,
+				type[strlen (type) -1] == '*' ? "": " ",
```
Comment (radare) [inline libr/core/disasm.c]: can be index -1

File: libr/core/disasm.c [hunk 2]
```diff
@@ -2837,13 +2864,17 @@ static void ds_print_calls_hints(RDisasmState *ds) {
...
 	int i, arg_max = r_anal_type_func_args_count (anal, name);
-	for (i = 0; i < arg_max; i++) {
-		char *type = r_anal_type_func_args_type (anal, name, i);
-		r_cons_printf ("%s%s%s%s%s", i == 0 ? "": " ", type,
-			type[strlen (type) -1] == '*' ? "": " ",
-			r_anal_type_func_args_name (anal, name, i),
-			i == arg_max - 1 ? ");\n": ",");
-		free (type);
+	if (arg_max == 0) {
+		r_cons_printf("void);\n");
```
Comment (radare) [inline libr/core/disasm.c]: missing space

File: libr/core/disasm.c [hunk 3]
```diff
@@ -1036,13 +1036,40 @@ static void ds_show_functions(RDisasmState *ds) {
...
 			r_list_foreach (args, iter, var) {
-				r_cons_printf ("%s %s%s", var->type, var->name, iter->n ? ", " : "");
+				if (var->delta > 0) {
+					if (!r_list_empty (regs) && comma) {
+						r_cons_printf (", ");
+						comma = false;
+					}
+					arg_bp = true;
+					r_cons_printf ("%s%s%s%s", var->type,
+						var->type[strlen (var->type) - 1] =='*' ? "" : " ",
```
Comment (radare) [inline libr/core/disasm.c]: index -1

File: libr/core/disasm.c [hunk 4]
```diff
@@ -1036,13 +1036,40 @@ static void ds_show_functions(RDisasmState *ds) {
...
-			//XXX probably broken
 			r_cons_printf ("%s%s%s %s %s%s (",
 				COLOR (ds, color_fline), ds->pre,
 				COLOR_RESET (ds), COLOR (ds, color_fname),
 				fcn_name, COLOR_RESET (ds));
+			bool comma = true;
+			bool arg_bp = false;
+			r_list_foreach (regs, iter, var) {
+				r_cons_printf ("%s%s%s%s", var->type,
+					var->type[strlen (var->type) - 1] =='*' ? "" : " ",
```
Comment (radare) [inline libr/core/disasm.c]: index can be -1

**Comment by oddcoder** [comment]
They are all fixed, github fault

**Comment by oddcoder** [comment]
Fixing broken tests here 
https://github.com/radare/radare2-regressions/pull/546

**Pull request merged**